rust_src/src/editfns.rs

```diff
@@ -182,3 +182,43 @@ pub fn char_after(mut pos: LispObject) -> LispObject {
         }
     }
 }
+
+/// Return a copy of STRING with text properties added.
+/// First argument is the string to copy.
+/// Remaining arguments form a sequence of PROPERTY VALUE pairs for text
+/// properties to add to the result.
+/// usage: (propertize STRING &rest PROPERTIES)  */
+#[lisp_fn(min = "1")]
+pub fn propertize(args: &mut [LispObject]) -> LispObject {
+    /* Number of args must be odd. */
+    if args.len() & 1 == 0 {
+        error!("Wrong number of arguments");
+    }
+
+    let mut it = args.iter();
+
+    let first = it.next().unwrap(); // safe, there is at least 1 arg
+    if !first.is_string() {
+        wrong_type!(Qstringp, *first);
+    }
+
+    let copy = first.clone();
+    let mut properties = Qnil;
```

Comment:
We prefer to use LispObject::constant_nil() over a direct usage of Qnil.

Reply:
I did not because properties is always a C Lisp_Object, not a Rust LispObject. This leads to type errors.

Reply:
I can add a comment so people see it. Eventually this will go away once Fcons is ported.